core/sequencer/server_test.go

```diff
@@ -229,3 +252,68 @@ func TestHighWatermarks(t *testing.T) {
 		})
 	}
 }
+
+func TestDuplicateUpdates(t *testing.T) {
+	ctx := context.Background()
+	initMetrics.Do(func() { createMetrics(monitoring.InertMetricFactory{}) })
+	ks, err := keyset.NewHandle(signature.ECDSAP256KeyTemplate())
+	if err != nil {
+		t.Fatalf("signature.NewHandle(): %v", err)
+	}
+	signer, err := signature.NewSigner(ks)
+	if err != nil {
+		t.Fatalf("signature.NewSigner(): %v", err)
+	}
+	authorizedKeys, err := ks.Public()
+	if err != nil {
+		t.Fatalf("Failed to setup tink keyset: %v", err)
+	}
+
+	index := []byte("index")
+	userID := "userID"
+	log0 := []mutator.LogMessage{}
+	mapRev := int64(0)
+	for i, data := range []string{"data1", "data2"} {
+		m := entry.NewMutation(index, directoryID, userID)
+		if err := m.SetCommitment([]byte(data)); err != nil {
+			t.Fatalf("%v", err)
+		}
+		if err := m.ReplaceAuthorizedKeys(authorizedKeys.Keyset()); err != nil {
+			t.Fatalf("%v", err)
+		}
+		update, err := m.SerializeAndSign([]tink.Signer{signer})
+		if err != nil {
+			t.Fatalf("SerializeAndSign(): %v", err)
+		}
+		log0 = append(log0, mutator.LogMessage{
+			ID:        int64(i),
+			Mutation:  update.Mutation,
+			ExtraData: update.Committed},
+		)
+	}
+
+	s := Server{
```

Comment:
It seems like the only reason to create `Server` this way is that a non-default `trillian` field can't be initialized in the constructor. How about calling `NewServer`, and updating the `trillian` field post-factum? Also, this would allow removing line 258.
Alternatively, maybe add some shorter non-exported `newServerImpl` method to `Server`? The `NewServer` method would call that one.